### PR TITLE
add basic documentation for all plugins

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
Test failure looks unrelated.

File: doc/manual/src/plugins.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+- `enable_bitbucket_status`
+- `bitbucket.username`
+- `bitbucket.password`
+
+## Circle CI Notifcation
+
+Sets Circle CI status
+
+configuration options:
+- `circleci.[].jobs`
```
Comment (Mic92) [inline doc/manual/src/plugins.md:36]: I am open for suggestions regarding the notation. Some options can be repeated, some take a key, others are scalar.

File: doc/manual/src/plugins.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+- `githubstatus.[].inputs`
+
+The input which corresponds to the github repo/rev whose
+status we want to report. Can be repeated.
+
+- `githubstatus.[].authorization`
+
+Verbatim contents of the Authorization header. See
+[GitHub documentaion](https://developer.github.com/v3/#authentication) for
```
Comment (Mic92) [inline doc/manual/src/plugins.md:134]:
```suggestion
[GitHub documentation](https://developer.github.com/v3/#authentication) for
```

File: doc/manual/src/plugins.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
+# Plugins
+
+This chapter describes all plugins present in hydra
```
Comment (grahamc) [inline doc/manual/src/plugins.md:3]:
```suggestion
This chapter describes all plugins present in Hydra.
```

File: doc/manual/src/plugins.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+### Inputs
+
+Hydra supports the following inputs
```
Comment (grahamc) [inline doc/manual/src/plugins.md:7]:
```suggestion
Hydra supports the following inputs:
```

File: doc/manual/src/plugins.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+- Git input
+- Mercurial input
+- Path input
+
+## Bitbucket pull requests
+
+Create jobs based on open bitbucket pull requests
+
+configuration options:
+- `bitbucket_authorization.<owner>`
```
Comment (grahamc) [inline doc/manual/src/plugins.md:20]: In general, what do you think about:

* making the first line a full sentence and adding a period at the end?
* making `configuration options:` a header?

Like this:

```suggestion
## Bitbucket pull requests

Create jobs based on open Bitbucket pull requests.

### Configuration Options
- `bitbucket_authorization.<owner>`
```

File: doc/manual/src/plugins.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+Sets Bitbucket CI status
+
+configuration options:
+- `enable_bitbucket_status`
+- `bitbucket.username`
+- `bitbucket.password`
+
+## Circle CI Notifcation
+
+Sets Circle CI status
```
Comment (grahamc) [inline doc/manual/src/plugins.md:33]: (product name)

```suggestion
## CircleCI Notifcation

Sets CircleCI status
```

File: doc/manual/src/plugins.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+- `coverityscan.[].jobs`
+- `coverityscan.[].project`
+- `coverityscan.[].email`
+- `coverityscan.[].token`
+- `coverityscan.[].scanurl`
+
+## Declarative jobsets
+
+TODO
```
Comment (grahamc) [inline doc/manual/src/plugins.md:70]: see the top level review comment

File: doc/manual/src/plugins.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+Sets Gitea CI status
+
+configuration options:
+
+- `gitea_authorization.<repo-owner>`
+
+## Github pulls
+
+Create jobs based on open github pull requests
```
Comment (grahamc) [inline doc/manual/src/plugins.md:90]: (product name)

```suggestion
## GitHub pulls

Create jobs based on open GitHub pull requests
```

There are several other capitalizations of GitHub below (GitHub, Github, github) and they should probably all be GitHub, except for `code blocks` and URLs.

File: doc/manual/src/plugins.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+```xml
+<githubstatus>
+  jobs = test:pr:build
+  inputs = src
+  authorization = Basic notgivingyoumypasswordosorry
+  excludeBuildFromContext = 1
+</githubstatus>
+```
+
+## Gitlab pulls
```
Comment (grahamc) [inline doc/manual/src/plugins.md:157]: (product name)

```suggestion
## GitLab pulls
```

There are similarly different capitalizations below too.

File: doc/manual/src/plugins.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+```
+ file sotest-binaries /nix/store/…/binaries.zip
+ file sotest-config /nix/store/…/config.yaml
+```
+
+### Configuration options:
+
+- `sotest.[].uri`
+
+Url of the controlle, defaults to `https://opensource.sotest.io`
```
Comment (grahamc) [inline doc/manual/src/plugins.md:259]:
```suggestion
URL of the controller, defaults to `https://opensource.sotest.io`
```

File: doc/manual/src/plugins.md [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+
+- `runcommand.[].job`
+
+Regular expression for jobs to match in the format `project:jobset:job`.
+Defaults to `*:*:*`.
+
+- `runcommand.[].command`
+
+Command to run. Can use the `$HYDRA_JSON` environment variable to access
+information about the buiild.
```
Comment (grahamc) [inline doc/manual/src/plugins.md:207]:
```suggestion
information about the build.
```

File: doc/manual/src/plugins.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+## Bitbucket status
+
+Sets Bitbucket CI status
+
+configuration options:
+- `enable_bitbucket_status`
+- `bitbucket.username`
+- `bitbucket.password`
+
+## Circle CI Notifcation
```
Comment (grahamc) [inline doc/manual/src/plugins.md:31]:
```suggestion
## Circle CI Notification
```

**Review by grahamc — COMMENTED** [review]
This is great, thank you a lot!

What would you think about moving this file to `plugins/README.md`? This would let us move the Declarative Jobsets documentation in to `./plugins/declarative-jobsets.md` and link to it from the todo'd area.

**Comment by Mic92** [comment]
I moved declarative projects to plugins.

**Pull request merged**